dali/pipeline/data/tensor_vector.cc

```diff
@@ -233,8 +260,8 @@ void TensorVector<Backend>::reserve(size_t bytes_per_sample, int batch_size) {
   assert(batch_size > 0);
   state_ = State::noncontiguous;
   resize_tensors(batch_size);
-  for (auto t : tensors_) {
-    t->reserve(bytes_per_sample);
+  for (size_t i = 0; i < curr_tensors_size_; i++) {
```

Comment:
reserves allows to reserve that many tensors as `batch_size` provided. If there is less tensors it should allocate more and reserve each of them to the desired state.

Reply:
Ok resize_tensors adjust the curr_tenstors_size_.